envs/quadrotors/quad_multi_model.py

```diff
@@ -0,0 +1,60 @@
+import torch
+from torch import nn
+
+from algorithms.appo.model_utils import get_obs_shape, nonlinearity, create_standard_encoder, EncoderBase, \
+    register_custom_encoder
+from algorithms.utils.pytorch_utils import calc_num_elements
+from utils.utils import log
+
+
+class QuadMultiMeanEncoder(EncoderBase):
+    '''Mean embedding encoder based on the DeepRL for Swarms Paper'''
+    def __init__(self, cfg, obs_space, timing, self_obs_dim=18, neighbor_obs_dim=6, neighbor_hidden_size=32):
+        super().__init__(cfg, timing)
+        self.self_obs_dim = self_obs_dim
+        self.neighbor_obs_dim = neighbor_obs_dim
+        self.neighbor_hidden_size = neighbor_hidden_size
+        self.device = torch.device('cpu' if cfg.device == 'cpu' else 'cuda')
```

Comment:
This is not going to work with multiple GPUs, 'cuda' means GPU #0, in a multi-GPU system you need to pass an index as well, like `cuda:0`

Also, Zhehui is right, I don't think you need self.device at all.
And if you need it in forward function, you can always use obs.device